src/floris/simulation/wake.py

```diff
@@ -13,233 +13,132 @@
-            else:
-                self._turbulence_model = self._turbulence_models[value](
-                    self.parameters["wake_turbulence_parameters"]
-                )
-        elif isinstance(value, WakeTurbulence):
-            self._turbulence_model = value
-        else:
-            raise ValueError("Invalid value given for WakeTurbulence: {}".format(value))
-
-    @property
-    def deflection_model(self):
-        """
-        Deflection model.
-
-        **Note**: This is a virtual property used to "get" or "set" a value.
-
-        Args:
-            value (str, :py:class:`~.base_velocity_deflection.VelocityDeflection`):
-                A string for the model to set or the model instance itself.
-
-        Returns:
-            :py:class:`~.base_velocity_deflection.VelocityDeflection`:
-                Model currently set.
+    Args:
+        wake (:obj:`dict`): The wake's properties input dictionary
+            - velocity_model (str): The name of the velocity model to be
+                instantiated.
+            - turbulence_model (str): The name of the turbulence model to be
+                instantiated.
+            - deflection_model (str): The name of the deflection model to be
+                instantiated.
+            - combination_model (str): The name of the combination model to
+                be instantiated.
+    """
 
-        Raises:
-            ValueError: Invalid value.
-        """
-        return self._deflection_model
-
-    @deflection_model.setter
-    def deflection_model(self, value):
-        if type(value) is str:
-            if "wake_deflection_parameters" not in self.parameters.keys():
-                self._deflection_model = self._deflection_models[value]({})
-            else:
-                self._deflection_model = self._deflection_models[value](
-                    self.parameters["wake_deflection_parameters"]
+    model_strings: dict
+    wake_combination_parameters: dict = attr.ib(factory=dict)
+    wake_deflection_parameters: dict = attr.ib(factory=dict)
+    wake_turbulence_parameters: dict = attr.ib(factory=dict)
+    wake_velocity_parameters: dict = attr.ib(factory=dict)
+
+    combination_model: Any = attr.ib(init=False)
+    deflection_model: Any = attr.ib(init=False)
+    turbulence_model: Any = attr.ib(init=False)
+    velocity_model: Any = attr.ib(init=False)
+    model_string: str = model_attrib(default="wake")
+
+    def __attrs_post_init__(self) -> None:
+        self.model_generator()
+
+    def model_generator(self) -> Any:
+        models = ("combination", "deflection", "turbulence", "velocity")
+        wake_models = {}
+        for model_type in models:
+            model_opts = MODEL_MAP[f"wake_{model_type}"]
+            try:
+                model_string = self.model_strings[f"{model_type}_model"]
+            except KeyError:
+                wake_models[model_type] = None
+                continue  # TODO: We don't have to create all model types?
```

Comment:
So it would raise an error if the user tried to select (or had a typo) a model that doesn't exist? I would agree with that.

Reply:
My thought here was we don't actually have any combination or turbulence models implemented yet, so we'd not be able to actually run Floris until that point, but otherwise I agree entirely. Should I add it as an issue to let it go to a `KeyError` at that point?

Reply:
I suggest let's comment out the things that require a turbulence and combination model until they're implemented. From the solver side, the models are in place but they aren't split into distinct classes like the velocity deficit and deflection models. Letting Python throw the error here makes sense to me.

Reply:
Sounds good, I'll update this to only check for model types we've implemented then.